### PR TITLE
Added nf-test for pbtk/bam2fastq

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/pbtk/bam2fastq/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
+    test("homo_sapiens - bam - stub - u") {
```
Comment (SPPearce) [inline modules/nf-core/pbtk/bam2fastq/tests/main.nf.test:60]: Can you test this without the stub too?

**Review by SPPearce — COMMENTED** [review]
Can you just test the uncompressed version without the stub please.

Also, I remember there being talk about deprecating another module in favour of this one.

**Comment by famosab** [comment]
I think this is all related to these PRs: #4798 #4797 and #4791 

Depending on who is asked the deprecation should either be done in one PR or in two PRs.

File: modules/nf-core/pbtk/bam2fastq/tests/main.nf.test (hunk 1 above)
Comment (GallVp) [reply modules/nf-core/pbtk/bam2fastq/tests/main.nf.test:85]: Good idea. Done. Thanks!

**Comment by GallVp** [comment]
> I think this is all related to these PRs: #4798 #4797 and #4791
> 
> Depending on who is asked the deprecation should either be done in one PR or in two PRs.

Thank you! I have added the deprecation notice here.

**Comment by SPPearce** [comment]
> I think this is all related to these PRs: #4798 #4797 and #4791 
> 
> Depending on who is asked the deprecation should either be done in one PR or in two PRs.

Eh, I don't care whether it is in one PR or two. The only reason for me to split things up is to make it easier for reviewers

**Comment by GallVp** [comment]
Thanks @SPPearce 

So we should close all the related PRs: https://github.com/nf-core/modules/pull/4798 https://github.com/nf-core/modules/pull/4797 and https://github.com/nf-core/modules/pull/4791

**Pull request merged**